example/lib/main.dart

```diff
@@ -1,112 +1,111 @@
+// ignore_for_file: public_member_api_docs, avoid_redundant_argument_values
+
 import 'package:flutter/material.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'package:freerasp/freerasp.dart';
-import 'package:freerasp_example/safety_icon.dart';
 import 'package:freerasp_example/threat_notifier.dart';
+import 'package:freerasp_example/threat_state.dart';
+import 'package:freerasp_example/widgets/widgets.dart';
 
 /// Represents current state of the threats detectable by freeRASP
-final threatProvider = StateNotifierProvider<ThreatNotifier, Map<Threat, bool>>(
-  (ref) => ThreatNotifier(),
-);
+final threatProvider =
+    NotifierProvider.autoDispose<ThreatNotifier, ThreatState>(() {
+  return ThreatNotifier();
+});
 
-void main() async {
-  /// Make sure that bindings are initialized before using Talsec.
+Future<void> main() async {
   WidgetsFlutterBinding.ensureInitialized();
 
+  /// Initialize Talsec config
+  await _initializeTalsec();
+
+  runApp(const ProviderScope(child: App()));
+}
+
+/// Initialize Talsec configuration for Android and iOS
+Future<void> _initializeTalsec() async {
   final config = TalsecConfig(
-    /// For Android
     androidConfig: AndroidConfig(
       packageName: 'com.aheaditec.freeraspExample',
       signingCertHashes: ['AKoRuyLMM91E7lX/Zqp3u4jMmd0A7hH/Iqozu0TMVd0='],
       supportedStores: ['com.sec.android.app.samsungapps'],
+      blocklistedPackageNames: ['com.aheaditec.freeraspExample'],
```

Comment:
I think it would be nicer to have the malware config separated from other fields
```suggestion
      malware: MalwareConfig(
          blocklistedPackageNames: ['com.aheaditec.freeraspExample'],
      ),
```